backend/src/main/backend/Table.java

```diff
@@ -0,0 +1,100 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+public class Table {
+    private String name;
+
+    private Connection mConnection;
+    private PreparedStatement mCreateTable;
+    private PreparedStatement mDropTable;
+    private PreparedStatement mDeleteRow;
+    private PreparedStatement mInsertRow;
+    private PreparedStatement mUpdateRow;
+    // Other prepared statements...
+
+    public Table(String tName){
+        name = tName;
+    }
+
+    public String getName(){
+        return name;
+    }
+
+    public void create(Database db) {
+        try {
+            if(name.equals("tblMessages")) {
+                db.mCreateTable = db.mConnection.prepareStatement("CREATE TABLE IF NOT EXISTS tblMessages (row_id SERIAL PRIMARY KEY, subject VARCHAR(50) NOT NULL, content VARCHAR(500) NOT NULL, user_id INTEGER REFERENCES tblUsers)");
+            }
+            else if(name.equals("tblComment")) {
+                db.mCreateTable =  db.mConnection.prepareStatement("CREATE TABLE IF NOT EXISTS tblComment (row_id SERIAL PRIMARY KEY, message_id INTEGER REFERENCES tblMessages, user_id INTEGER REFERENCES tblUsers, content VARCHAR(500) NOT NULL)");
+            }
+            else if(name.equals("tblUsers")) {
+                db.mCreateTable = db.mConnection.prepareStatement("CREATE TABLE IF NOT EXISTS tblUsers (row_id SERIAL PRIMARY KEY, user_name VARCHAR(500) NOT NULL, first_name VARCHAR(500) NOT NULL, last_name VARCHAR(500) NOT NULL, email VARCHAR(500) NOT NULL)");
+            }
+            else if(name.equals("tblLikes")) {
+                db.mCreateTable = db.mConnection.prepareStatement("CREATE TABLE IF NOT EXISTS tblLikes (row_id SERIAL PRIMARY KEY, user_id INTEGER REFERENCES tblUsers, message_id INTEGER REFERENCES tblMessages, content INTEGER)");    
+            }
+
+            db.mCreateTable.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void drop(Database db) {
+        String statement = "DROP TABLE IF EXISTS " + name + " CASCADE;"; // Added space before CASCADE
+        try {
+            db.mDropTable = db.mConnection.prepareStatement(statement);
+            db.mDropTable.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public int insertRow(Database db, ArrayList<String> content) {
+        StringBuilder statement = new StringBuilder("INSERT INTO " + name + " VALUES (default");
+
+        for (int i = 0; i < content.size(); i++) {
+            statement.append(", ?");
```

Comment:
This will work just fine, but I also suggest taking a look at the `StringJoiner` class that builds on top of `StringBuilder` - it has made my life so much nicer when doing these kinds of operations.

Reply:
just read up the documentation on this. very interesting! never knew of this class. this is how i have it now. what do you think?:

```
StringJoiner statement = new StringJoiner(", ", "INSERT INTO " + name + " VALUES (default,", ")");
for (int i = 0; i < content.size(); i++) {
    statement.add(" ?");
}
```